core/jvm/src/main/scala/zio/internal/metrics/ConcurrentMetricHooksPlatformSpecific.scala

```diff
@@ -174,7 +174,7 @@ private[zio] class ConcurrentMetricHooksPlatformSpecific extends ConcurrentMetri
     // While Observing we cut off the first sample if we have already maxSize samples
     def observe(tuple: (Double, java.time.Instant)): Unit = {
       if (maxSize > 0) {
-        val target = head.incrementAndGet() % maxSize
+        val target = head.updateAndGet(index => (index + 1) % maxSize)
```

Comment:
I think it may be better just to make this an `AtomicLong`.

Reply:
That works too, updated, thanks.